bin/gencode_buckets

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+""" Gencode generator for buckets """
+import re
+import os
+
+GENCODE_MARKER = '@@ '
+BUCKET_MARKER = '* '
+WILDCARD_REGEX = r' *\* _\?\?\?_: (.*)'
+WILDCARD_SUFFIX = r'.[a-z]+(_[a-z]+)*'
+
+BUCKET_REGEX = r' *\* _([a-z]+)_: (.*)'
+JSON_FORMAT = '%s%s{ "pattern": "^%s$" }'
+DEVICE_PREFIX = 'device'
+
+JAVA_DESCRIPTION = '\n%s// %s\n'
+JAVA_TARGET = '%s%s("%s")%s\n'
+
+doc_in = os.path.join('docs/specs/buckets.md')
```

Comment:
Make doc_in, java_in, java_out CONSTANTS, or, move them to main() and pass them as arguments to the functions (better for testing too). Prefer latter

Also, not sure on the need for os.path.join() ?

Reply:
Did both actually (since they're CONSTANTS that should be passed in). And removed os.path.join()